include/dynd/kernels/arithmetic.hpp

```diff
@@ -68,7 +68,49 @@ namespace nd {
                                          &detail::inline_logical_xor<Src0TypeID, Src1TypeID>::f> {
   };
 
-#undef DYND_DeclBinopKernel
+#define DYND_DEF_BINARY_OP_KERNEL_ZEROCHECK_INT(OP, NAME)                                                              \
+  namespace detail {                                                                                                   \
+    template <type_id_t Src0TypeID, type_id_t Src1TypeID,                                                              \
+              type_id_t Src0Base = base_id_of<Src0TypeID>::value,                                                      \
+              type_id_t Src1Base = base_id_of<Src1TypeID>::value,                                                      \
+              bool check = ((Src0Base == bool_kind_id) || (Src0Base == int_kind_id) || (Src0Base == uint_kind_id)) &&  \
+                           ((Src1Base == bool_kind_id) || (Src1Base == int_kind_id) || (Src1Base == uint_kind_id))>    \
+    struct inline_##NAME;                                                                                              \
+                                                                                                                       \
+    template <type_id_t Src0TypeID, type_id_t Src1TypeID, type_id_t Src0Base, type_id_t Src1Base>                      \
+    struct inline_##NAME<Src0TypeID, Src1TypeID, Src0Base, Src1Base, true> {                                           \
+      typedef typename type_of<Src0TypeID>::type T0;                                                                   \
+      typedef typename type_of<Src1TypeID>::type T1;                                                                   \
+      typedef decltype(std::declval<T0>() OP std::declval<T1>()) R;                                                    \
+      static R f(T0 a, T1 b) {                                                                                         \
+        if(b == 0) {                                                                                                   \
+          throw dynd::zero_division_error("Integer division or modulo by zero.");                                      \
+        }                                                                                                              \
+        return a OP b;                                                                                                 \
+      }                                                                                                                \
+    };                                                                                                                 \
+                                                                                                                       \
+    template <type_id_t Src0TypeID, type_id_t Src1TypeID, type_id_t Src0Base, type_id_t Src1Base>                      \
```

Comment:
I did that deliberately since the code is completely identical for unsigned integers, signed integers, and bools and we have to check that both arguments are one of those types. That's why I originally implemented this with `std::is_integral`. Is there a another way to do this that doesn't involve rewriting the same thing nine times?